src/test/java/uk/ac/cam/cl/dtg/segue/api/AdminFacadeIT.java

```diff
@@ -0,0 +1,113 @@
+
+import static org.easymock.EasyMock.createMock;
+import static org.easymock.EasyMock.createNiceMock;
+import static org.easymock.EasyMock.replay;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static uk.ac.cam.cl.dtg.isaac.api.ITConstants.DELETION_TEST_STUDENT_EMAIL;
+import static uk.ac.cam.cl.dtg.isaac.api.ITConstants.DELETION_TEST_STUDENT_ID;
+import static uk.ac.cam.cl.dtg.isaac.api.ITConstants.TEST_ADMIN_ID;
+import static uk.ac.cam.cl.dtg.isaac.api.ITConstants.TEST_NON_EXISTENT_USER_ID;
+import static uk.ac.cam.cl.dtg.isaac.api.ITConstants.TEST_TEACHER_EMAIL;
+import static uk.ac.cam.cl.dtg.isaac.api.ITConstants.TEST_TEACHER_PASSWORD;
+
+import jakarta.servlet.http.Cookie;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.ws.rs.core.Response;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import uk.ac.cam.cl.dtg.isaac.api.IsaacIntegrationTest;
+import uk.ac.cam.cl.dtg.isaac.dos.users.RegisteredUser;
+import uk.ac.cam.cl.dtg.segue.api.managers.IExternalAccountManager;
+import uk.ac.cam.cl.dtg.segue.api.managers.StatisticsManager;
+import uk.ac.cam.cl.dtg.segue.auth.exceptions.AdditionalAuthenticationRequiredException;
+import uk.ac.cam.cl.dtg.segue.auth.exceptions.AuthenticationProviderMappingException;
+import uk.ac.cam.cl.dtg.segue.auth.exceptions.IncorrectCredentialsProvidedException;
+import uk.ac.cam.cl.dtg.segue.auth.exceptions.MFARequiredButNotConfiguredException;
+import uk.ac.cam.cl.dtg.segue.auth.exceptions.NoCredentialsAvailableException;
+import uk.ac.cam.cl.dtg.segue.dao.SegueDatabaseException;
+import uk.ac.cam.cl.dtg.segue.scheduler.SegueJobService;
+
+class AdminFacadeIT extends IsaacIntegrationTest {
+  private AdminFacade adminFacade;
+
+  @BeforeEach
+  void beforeEach() {
+    // These three classes are only used in singular methods that we don't have tests for currently,
+    // so just mock them for now
+    StatisticsManager statisticsManager = createMock(StatisticsManager.class);
+    SegueJobService segueJobService = createMock(SegueJobService.class);
+    IExternalAccountManager externalAccountManager = createMock(IExternalAccountManager.class);
+    this.adminFacade = new AdminFacade(properties, userAccountManager, contentManager, logManager, statisticsManager,
+        userPreferenceManager, eventBookingManager, segueJobService, externalAccountManager, misuseMonitor,
+        emailManager);
+  }
+
+  @Nested
+  class DeleteUserAccount {
+    @Test
+    void piiIsCorrectlySanitised() throws SegueDatabaseException {
+      HttpServletRequest mockRequest = prepareAdminRequest();
+
+      try (Response response = adminFacade.deleteUserAccount(mockRequest, DELETION_TEST_STUDENT_ID)) {
```

Comment:
does the deletable test student actually have a DoB? If not, I'm not certain you're testing the broken case?

Reply:
Ah, you appear to be correct that they don't have one there. Missed that only some of the existing test users had one set when I was copying values for the new entry. I'll correct that and check that everything still works.